src/metabase/sync/concurrent.clj

```diff
@@ -16,7 +16,7 @@
 (defonce ^:private executor
   (delay (Executors/newFixedThreadPool 1 ^ThreadFactory thread-factory)))
 
-(defn submit-task
+(defn submit-task!
```

Comment:
A little on the fence about this because calling this 100 times with a pure function is technically pure even if calling it 100 times is absolutely not the same as calling it once. Most of our other `submit!` functions are `!`y so I followed that convention.